src/client/game.cpp

```diff
@@ -473,6 +474,11 @@ class GameGlobalShaderConstantSetter : public IShaderConstantSetter
 		get_sunlight_color(&sunlight, daynight_ratio);
 		m_day_light.set(sunlight, services);
 
+		auto lighting = m_client->getEnv().getLocalPlayer()->getLighting();
```

Comment:
I think it is already not copied and returned as a reference (it can be seen in the prototype definition). And `auto` should return the reference also as a part of the type.

Reply:
> And auto should return the reference also as a part of the type.

As far as I know, this is wrong. According to [that page](https://blog.petrzemek.net/2017/12/08/when-auto-seemingly-deduces-a-reference-in-cpp/), a bare auto is usually a copy.

Reply:
a

Reply:
a